src/mt.rs

```diff
@@ -171,21 +189,19 @@ impl Drop for MT {
     }
 }
 
-/// A meta-tracer aware thread.
+/// A meta-tracer aware thread. Note that this is conceptually a "front-end" to the actual
+/// meta-tracer thread akin to an `Rc`: this struct can be freely `clone()`d without duplicating
+/// the underlying meta-tracer thread. Note that this struct is neither `Send` nor `Sync`: it
+/// can only be accessed from within a single thread.
+#[derive(Clone)]
 pub struct MTThread {
-    mt: Arc<Mutex<MT>>,
-    hot_threshold: HotThreshold,
+    inner: Rc<MTThreadInner>,
```

Comment:
The use of `Rc` here is actually quite clever and subtle. I didn't appreciate this on first read.

Reply:
Yeah, it's kind of the core of the idea. It also enforces indirectly that `MTThread` isn't `Send` or `Sync`. Still, it's possible in the future that we might need to change to an `Arc` (perhaps, say, because we need some way for `MT` to directly mutate `MTThreadInner`) but at least the external guarantees we make will still hold.

Reply:
Let's see how we go. Hopefully we can get away with delegating mutation to the inner.

Reply:
If all else fails, we can always delegate mutation to an inner inner (e.g. using `Cell`).